internal/integration/email_test.go

```diff
@@ -0,0 +1,89 @@
+package integration
+
+import (
+	"context"
+	"database/sql"
+	"testing"
+
+	"github.com/fdemchenko/exchanger/internal/repositories"
+	"github.com/fdemchenko/exchanger/internal/services"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"github.com/testcontainers/testcontainers-go/modules/postgres"
+)
+
+type EmailServiceSuite struct {
+	suite.Suite
+	emailService services.EmailService
+	container    *postgres.PostgresContainer
+}
+
+func (em *EmailServiceSuite) SetupSuite() {
+	t := em.T()
+	container, err := createTestDBContainer()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	em.container = container
+	dsn, err := container.ConnectionString(context.Background(), "sslmode=disable")
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Log(dsn)
+	db, err := sql.Open("postgres", dsn)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	repo := &repositories.PostgresEmailRepository{DB: db}
+	em.emailService = services.NewEmailService(repo)
+}
+
+func (em *EmailServiceSuite) TearDownTest() {
+	err := em.container.Restore(context.Background())
+	if err != nil {
+		em.T().Fatal(err)
+	}
+}
+
+func (em *EmailServiceSuite) TestCreateEmail_Success() {
+	err := em.emailService.Create("someemail@gmail.com")
+	assert.NoError(em.T(), err)
+}
+
+func (em *EmailServiceSuite) TestCreateEmail_Duplicate() {
+	t := em.T()
+	err := em.emailService.Create("somemail@gmail.com")
+	assert.NoError(t, err)
+
+	err = em.emailService.Create("somemail@gmail.com")
+	assert.ErrorIs(t, err, repositories.ErrDuplicateEmail)
+}
```

Comment:
These tests can be merged into one with two test cases.
Consider using [table test](https://dave.cheney.net/2019/05/07/prefer-table-driven-tests) in such cases